File: src/main/kotlin/social/alone/server/push/FcmTokenRegisterSvc.java
package social.alone.server.push;

import lombok.RequiredArgsConstructor;
import org.springframework.stereotype.Service;
import social.alone.server.user.User;

import java.util.Optional;

@Service
@RequiredArgsConstructor
public class FcmTokenRegisterSvc {

    private final FcmTokenRepository fcmTokenRepository;

    FcmToken register(String token, Optional<User> user) {
        return fcmTokenRepository.save(new FcmToken(token, user));
    }
}

File: src/main/kotlin/social/alone/server/push/FcmTokenRegisterSvc.kt
package social.alone.server.push

import org.springframework.beans.factory.annotation.Autowired
import org.springframework.stereotype.Service
import social.alone.server.push.domain.FcmToken
import social.alone.server.push.infra.FcmTokenRepository
import social.alone.server.user.domain.User

@Service
class FcmTokenRegisterSvc(@Autowired private val repository: FcmTokenRepository) {

    internal fun register(token: String, user: User?): FcmToken {
        repository.findByValue(token)?.let {
            it.updateUser(user)
            return it
        }
        return repository.save(FcmToken(token, user))
    }
}
